test/test_corpus.cc

```diff
@@ -601,6 +602,110 @@ TEST_P(LSPTest, All) {
+            auto fileUri = testFileUris[*filenames.begin()];
+            // TODO(sushain): stop manually copying these over
+            auto params = make_unique<TextDocumentCodeActionParams>(
+                make_unique<TextDocumentIdentifier>(fileUri),
+                make_unique<Range>(make_unique<Position>(error->range->start->line, error->range->start->character),
+                                   make_unique<Position>(error->range->end->line, error->range->end->character)),
+                make_unique<CodeActionContext>(move(diagnostics)));
+            auto req = make_unique<RequestMessage>("2.0", nextId++, LSPMethod::TextDocumentCodeAction, move(params));
+            auto responses = lspWrapper->getLSPResponsesFor(LSPMessage(move(req)));
+            EXPECT_EQ(responses.size(), 1) << "Did not receive a response for a codeAction request.";
+
+            if (responses.size() == 1) {
+                auto &msg = responses.at(0);
+                EXPECT_TRUE(msg->isResponse());
+                if (msg->isResponse()) {
+                    auto &response = msg->asResponse();
+                    ASSERT_TRUE(response.result) << "Code action request returned error: " << msg->toJSON();
+                    auto &receivedCodeActionResponse =
+                        get<variant<JSONNullObject, vector<unique_ptr<CodeAction>>>>(*response.result);
+
+                    EXPECT_FALSE(get_if<JSONNullObject>(&receivedCodeActionResponse));
+
+                    if (!get_if<JSONNullObject>(&receivedCodeActionResponse)) {
+                        UnorderedMap<string, unique_ptr<CodeAction>> receivedCodeActionsByTitle;
+                        auto &receivedCodeActions = get<vector<unique_ptr<CodeAction>>>(receivedCodeActionResponse);
+                        for (auto &codeAction : receivedCodeActions) {
+                            bool codeActionTitleUnique =
+                                receivedCodeActionsByTitle.find(codeAction->title) == receivedCodeActionsByTitle.end();
+                            EXPECT_TRUE(codeActionTitleUnique)
+                                << "Found code action with duplicate title: " << codeAction->title;
+
+                            if (codeActionTitleUnique) {
+                                receivedCodeActionsByTitle[codeAction->title] = move(codeAction);
+                            }
+                        }
+
+                        vector<shared_ptr<ApplyCodeActionAssertion>> matchedCodeActionAssertions;
+
+                        // Check any code action assertions for the same error range, i.e. verify that when applied,
+                        // they match the expectation.
+                        auto it = applyCodeActionAssertions.begin();
+                        while (it != applyCodeActionAssertions.end()) {
+                            auto codeActionAssertion = it->get();
+                            // TODO(sushain): what if there are multiple errors on the same range?
```

Comment:
This does happen in a few cases. We also have an issue with duplicate errors for some types of errors, so watch out... (see `error-with-dupes` annotation in tests).